### PR TITLE
Refactor Rock-on contribution from repository to Rockstor Documentation

### LINKED ISSUE #451: Refactor Rock-on Contribution and related sections
As per issue:
https://github.com/rockstor/rockon-registry/issues/350

move contribution instructions and guidelines to mainline Rockstor documentation to follow the canonical structure as outlined in the above Issue.

This involves changing:
https://rockstor.com/docs/contribute/contribute_rockons.html
https://rockstor.com/docs/interface/overview.html#adding-rockons

### PR DESCRIPTION
**Pull request by Hooverdan96**
Fixes #451.

### This pull request's proposal
 Changes to be committed:
	modified:   contribute/contribute_rockons.rst
	modified:   interface/overview.rst
- refactor, move and include contribution instructions from Rock-on repository into Rockstor documentation.
- Minor spelling/formulation corrections in both files

### Checklist
- [X] With the proposed changes no Sphinx errors or warnings are generated.
- [X] I have added my name to the AUTHORS file, if required (descending alphabetical order).

### CODE REVIEW

**Comment by Hooverdan96** [comment]
There are now 3 `403` errors left during linkcheck - these seem to be new, the website links work if I visit them manually. Suspect it might be a block due to "source" being github for hitting that link, or it's somewhat related to one of the issues I found while researching where the Sphinx/linkcheck  is having some issues for re-authorizing, due to the underlying python version and whatnot ...

**Comment by phillxnet** [comment]
@Hooverdan96 Thanks for seeing to this modification/normalisation re our contribution docs. Much appreciated.
Re:
> There are now 3 403 errors
Same here when run locally: I think we just find alternatives to these as all three have pop-up adverts in anyway. So likely we can find a more canonical source for these 2 nfs and 1 sssd-ad man pages.

**Comment by FroggyFlox** [comment]
Thanks a lot indeed, @Hooverdan96 !

With regards to finding a better link to manpages, may I suggest manpages.opensuse.org? Better suited to us and ad-free.

File: contribute/contribute_rockons.rst [hunk 1]
```diff
@@ -6,37 +6,76 @@ Contributing a Rock-on
...
+described here :ref:`rockons_intro`.
+
+2. Create your Rock-on profile file, [app].json following the clues on
+the Rock-on structure described in the Rock-on repository
+`readme.md <https://github.com/rockstor/rockon-registry/blob/master/README.md>`_.
+
+3. Upload the file to ``/opt/rockstor/rockons-metastore/[app].json``. Hit
+update in the Web-UI and install your brand new Rock-on!
+
+If you like to share your app with rest of the Rockstor community, open
```
Comment (phillxnet) [inline contribute/contribute_rockons.rst:29]:
```suggestion
If you would like to share your app with the rest of the Rockstor community, open
```
A little rewording suggestion if it takes your fancy :)

File: contribute/contribute_rockons.rst [hunk 2]
```diff
@@ -64,6 +103,98 @@ guidelines can help:
 * Finally, the image size is another helpful factor, with smaller images being
   usually favored.
 
+Environment setup
+-----------------
+
+Go to the `rockon-registry repo <https://github.com/rockstor/rockon-registry>`_
+and click on the <b>Fork</b> button. This will fork the repository into your
+profile which serves as your private git remote called origin. The next few
```
Comment (phillxnet) [inline contribute/contribute_rockons.rst:111]:
```suggestion
profile which serves as your personal git remote called origin. The next few
```
I had thought a default fork in this context was non private, ergo 'personal' suggested word change.

File: contribute/contribute_rockons.rst [hunk 3]
```diff
@@ -64,6 +103,98 @@ guidelines can help:
...
+
+Add a remote called upstream to periodically rebase your local repository
+with changes in the upstream made by other contributors.
+
+.. code-block:: console
+
+    git remote add upstream https://github.com/rockstor/rockon-registry.git
+
+The above 4 steps help you setup your local environment. If you are familiar
+with git and use an IDE like Eclipse, you can achieve the same outcome in a
```
Comment (phillxnet) [inline contribute/contribute_rockons.rst:137]:
```suggestion
with git and use an IDE, you can achieve the same outcome in a
```

Removing specific reference while I see it.

**Review by phillxnet — CHANGES_REQUESTED** [review]
@Hooverdan96 This is a great improvement. I was going to merge as-is but the html tags were the tipping point :). See what you think of the suggestions and I can pop this in directly there after hopefully.

Thanks for your ongoing work re improving our docs and code.

**Comment by Hooverdan96** [comment]
I guess I didn't check the text enough after copying it ... thanks for the additional eyes on it, and the rewording suggestions made sense as well.
I will create a separate issue/pull request to change the man-page targets, especially since those are in different parts of the documentation. @FroggyFlox thanks for the suggestion, the opensuse source makes even more sense to us now 💯

**Comment by Hooverdan96** [comment]
Once you're satisfied with the changes, I can squash them into one commit (though I know you mentioned that it's probably less relevant for the documentation).

**Review by phillxnet — APPROVED** [review]

**Comment by phillxnet** [comment]
@Hooverdan96 Thanks again. About to merge. Re squashing I think it's less important to squash in docs/rock-ons/website as eases review / additional attribution / contributor burden: and clean history is less important than in code repositories. Plus we have only one branch in these repositories. So cherry picking requirement from one branch to another is not a thing.

**Comment by phillxnet** [comment]
PR product PRODUCTION published.

**Pull request merged**